Add user guide and dev guide to main site

nicholaschuayunzhi opened the pull request:
Migrates user guide and developer guide to the website.

Preview is available [here](https://deploy-preview-3--markbind.netlify.com/)

**Things to note**
---
I have updated [developer guide](https://deploy-preview-3--markbind.netlify.com/developerguide) to reflect that `MarkBind` repo contains both CLI and core library.

The preview is only available if the changes in #186 are merged as seen in https://github.com/nicholaschuayunzhi/markbind/pull/3

damithc commented on docs/sections/includeAndRender.md:
```diff
@@ -0,0 +1,65 @@
+
+# Command: include and render
+
+MarkBind provides `include` and `render` commands to allow you include the "included" fragments in a file and render the file into a HTML document.
```
I wonder if include and render are really needed. As an author, I never use them. They are more useful for developers?

damithc commented on docs/sections/userQuickStart.md:
```diff
@@ -0,0 +1,131 @@
+- [x] Markdown
+- [x] HTML & CSS
+- [x] npm (Node.js)
+- [x] Command-line Environment
```
This list is a bit off-putting. May be no need for users to know html, css, node?

damithc commented on docs/sections/userQuickStart.md:
```diff
@@ -0,0 +1,131 @@
+`MarkBind` can be installed using npm
+
+`v1.4.3`: and above: Node.js with version 8.0 and higher is required.
+`v1.4.2`: We recommend users to use Node.js with version 6.5 or higher.
```
No need to document older versions. Future users will not be using those old versions anyway.

damithc commented on docs/sections/userQuickStart.md:
```diff
@@ -0,0 +1,131 @@
+- [x] npm (Node.js)
+- [x] Command-line Environment
+
+## Installation
```
Specifically mention how to install node, assuming the user doesn't know about node.

damithc commented on docs/sections/userQuickStart.md:
```diff
@@ -0,0 +1,131 @@
+
+You can add the help flag (`--help`) to any command to show the help screen.
+
+## Start Site Development
```
`Using MarkBind to Author a Website`?

damithc commented on docs/sections/userQuickStart.md:
```diff
@@ -0,0 +1,131 @@
+
+You may refer to this [doc]({{baseUrl}}/sections/siteConfiguration.html) for more details about how you can configure the `site.json`.
+
+### Authoring your contents.
```
remove full stop?

nicholaschuayunzhi commented:
thanks prof @damithc for the comments. I've updated the site and the changes are reflected here: https://deploy-preview-3--markbind.netlify.com/

I've also split the dev guide into smaller portions.

damithc commented:
Um, I thought we are keeping dev guide in markdown format because it needs to match latest master while  user guide should match the last release.

On another matter, I supposed we'll be migrating the vue strap part of the user guide as a separate PR?

nicholaschuayunzhi commented:
>  dev guide in markdown format because it needs to match latest master while user guide should match the last release.

Ah i see what you mean. So to clarify maybe this can be the workflow:
1. PR is made changes to the user guide in `docs` and also a changes to a dev guide (written in markdown)
1. PR is merged
    - dev guide is updated and changes are reflected on the repository
    - `docs` user guide is updated but not on the live site
1. On release, do `markbind deploy` to update the live site for the user guide

>  I supposed we'll be migrating the vue strap part of the user guide as a separate PR?

I was thinking of just providing a link for now as I had problems trying to get `affix` and the other components to work.

damithc commented:
Yes, that's the workflow I had in mind.

Integrating VueStrap docs can be done as a separate PR. If a component (e.g., affix) is not working in marbind format, that means we shouldn't include it in the user guide anyway :-) 
BTW, we don't need to follow VueStrap's current format of documenting components exactly. We can follow our own format, as long as it is clear to the reader.

rachx commented on docs/common/userGuideSections.md:
```diff
@@ -0,0 +1,7 @@
+- <a href="{{baseUrl}}/userGuide/userQuickStart.html">Quick Start</a>
+- <a href="{{baseUrl}}/userGuide/contentAuthoring.html">Content Authoring</a>
+- <a href="{{baseUrl}}/userGuide/developingASite.html">Developing a Site</a>
+- <a href="{{baseUrl}}/userGuide/ghpagesDeployment.html">Github Pages Deploment</a>
```
Deployment

rachx commented on docs/userGuide/contentAuthoring.md:
````diff
@@ -0,0 +1,92 @@
+
+To use an component, just type the correspond markup in your file. For example, to create a Panel, you just need to write:
+
+```
````
Should we add syntax highlighting (html) for this and other examples (including json)

rachx commented on docs/userGuide/contentAuthoring.md:
```diff
@@ -0,0 +1,92 @@
+
+Being able to include different markdown file into the current context is another feauture of *MarkBind*. You can create a complex document from different content fragments by including them.
+
+For detailed guide on using `<include>` tag for including contents, read the doc [here](includingContents.html).
```
Maybe we can use dynamic panels instead of linking to another page?
Not sure

nicholaschuayunzhi replied:
I'm not too sure about this. One thing is that we are leveraging on MarkBind's feature but then it seems unnecessary?

rachx commented on docs/userGuide/contentAuthoring.md:
````diff
@@ -0,0 +1,92 @@
+
+</Panel>
+```
+
````
Is this old?
I remembered we change it to mention `<md>` and `<markdown>` tags

nicholaschuayunzhi replied:
ah yes. I referred to an old version of the user guide. Will update accordingly!

damithc commented:
Let's merge this soon. The first version need not be perfect. We can improve it incrementally.

acjh commented:
@nicholaschuayunzhi You can merge this and #190 :)

damithc commented:
Can go ahead and do a deploy too? Unless this version of documentation has unreleased features.
